### PR TITLE
uncheck the empty folders and make them unclickable

### PR DESCRIPTION
**Pull request by irmakyz**

### CODE REVIEW

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@
         'geo-tree-item--single': isSingleItem
       }"
       :icon="categoryIcon"
-      @click="handleClick"
+      @click="(isSingleItem || !isEmptyCategory)? handleClick($event):()=>{}"
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:11]: Is better to modify only `handleClick`, right?

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 2]
```diff
@@ -243,8 +244,10 @@ export default {
 
       return !this.isChecked && isSomeChildSelected(this.category)
     },
+
     isChecked () {
       const allAreChildrenSelected = category => {
+        if (!_.isNil(category[this.keyForSubcategory]) && !_.size(category[this.keyForSubcategory])) return
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:250]: This should return a boolean, right?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 35.4.3
+New:
```
Comment (fofi) [inline CHANGELOG.md:4]: I think this is more a Fix

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 4]
```diff
@@ -256,11 +259,17 @@ export default {
...
+    isEmptyCategory () {
+      return !_.isNil(this.category[this.keyForSubcategory]) && !_.size(this.category[this.keyForSubcategory])
+    },
+
     totalCategoryChildren () {
       const sumOfChildren = category => _.size(category[this.keyForSubcategory]) + _.sumBy(category[this.keyForSubcategory], sumOfChildren)
 
       return sumOfChildren(this.category)
     },
+
```
Comment (Navas23) [inline src/elements/GeoTree/GeoTreeItem.vue:272]: I think in the DS we don't have empty lines in the component. We can decide if we wan't to add them from now on or not

Comment (dletran) [reply src/elements/GeoTree/GeoTreeItem.vue:272]: I'd say let's do it, IMO it helps readability as in the app

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 1 above)
Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:11]: - [ ] modify handleClick itself

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 2 above)
Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:250]: yess, hmm my fault i should do it different

Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:250]: - [ ] fix allAreChildrenSelected

File: CHANGELOG.md (hunk 3 above)
Comment (irmakyz) [reply CHANGELOG.md:4]: for the fixes don't we use patch ?

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 4 above)
Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:272]: i agree w david

File: CHANGELOG.md (hunk 3 above)
Comment (Navas23) [reply CHANGELOG.md:4]: He means the copy, to put Fix instead of New

Comment (irmakyz) [reply CHANGELOG.md:4]: - [ ] change copy of chagelog

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 5]
```diff
@@ -256,11 +260,17 @@ export default {
         ? allAreChildrenSelected(this.category)
         : !!this.checkedItems[this.category[this.keyForId]]
     },
+
+    isEmptyCategory () {
```
Comment (Navas23) [inline src/elements/GeoTree/GeoTreeItem.vue:264]: We could pass a category to this method and reuse it in `isChecked`

Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:264]: hmmm right, better idea! 

- [ ] pass category to isEmptyCategory and reuse

**Review by Navas23 — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 35.4.3
+Fixed:
```
Comment (fofi) [inline CHANGELOG.md:4]: We use "Fix", lets keep the same naming

File: src/elements/GeoTree/GeoTreeItem.vue [hunk 7]
```diff
@@ -297,19 +309,24 @@ export default {
      * On list item click
      */
     handleClick () {
+      if (!this.isSingleItem && this.isEmptyCategory(this.category)) return
```
Comment (fofi) [inline src/elements/GeoTree/GeoTreeItem.vue:312]: Why do we allow empty category if single item?

**Review by fofi — CHANGES_REQUESTED** [review]

File: src/elements/GeoTree/GeoTreeItem.vue (hunk 7 above)
Comment (fofi) [reply src/elements/GeoTree/GeoTreeItem.vue:312]: Then, why not add the return inside the else condition?

Comment (irmakyz) [reply src/elements/GeoTree/GeoTreeItem.vue:312]: hmmm yes better i am changing it

- [ ] check condition in else

**Review by fofi — APPROVED** [review]

**Review by dletran — APPROVED** [review]

**Pull request merged**